app/models/game.rb

```diff
@@ -42,13 +42,25 @@ def populate_board!
     ChessPiece.create(type: Queen, game_id: id, position_x: 3, position_y: 7, color: 'black')
     ChessPiece.create(type: King, game_id: id, position_x: 4, position_y: 7, color: 'black')
   end
+  scope :available, -> { where(black_player_id: nil).or(where(white_player_id: nil)) }
   
   
+  
+  def check?(color)
+    king = chess_pieces.find_by(type: 'King', color: color)
+    opponents = chess_pieces.find_by(color: color!)
+    
+    opponents.each do |piece|
+      if chess_piece.valid_move?(king.x_position, king.y_position)
```

Comment:
- The .valid_move? method should take in the board and the destination. At least that's how the method has been formatted in each of the piece models so far 
- Piece attributes should be .position_x and .position_y
- If I'm reading the logic correctly, you're checking to see if the current opponent's piece you're looping through can make a valid move on your king piece. So I think you could make an array of stop = [king.position_x, king.position_y] and pass that through as the second argument. You'll still need a 2D array of the board to pass through as well